lib/synapse/flat_file.rb

```diff
@@ -0,0 +1,53 @@
+require 'synapse/log'
+require 'fileutils'
+require 'tempfile'
+
+module Synapse
+  class FlatFile
+    include Logging
+    attr_reader :opts, :name
+
+    def initialize(opts)
+      super()
+
+      unless opts.has_key?("output_directory")
+        raise ArgumentError, "flat file generation requires an output_directory key"
+      end
+
+      begin
+        FileUtils.mkdir_p(opts['output_directory'])
```

Comment:
I figured that if synapse has permissions to write the directory, it should just go for it instead of erroring out. I use mkdir_p so that it'll be ok even if it is already created. Basically I think it leads to fewer preventable synapse errors.